src/api/spec/controllers/webui/comments_controller_spec.rb

```diff
@@ -105,6 +105,98 @@
       it { expect(flash[:success]).to eq('Comment deleted successfully.') }
       it { expect(Comment.where(id: other_comment.id)).to eq([]) }
     end
+
+    context 'whith the request_show_redesign beta flag active' do
+      render_views
+
+      before { Flipper.enable(:request_show_redesign, admin) }
+
+      let(:root_comment) { create(:comment_request, body: 'This is a root comment') }
+
+      context 'being a root comment' do
+
+        context 'with no replies' do
+          before do
+            root_comment
+
+            login admin
+            delete :destroy, params: { id: root_comment.id }
+          end
+
+          it 'removes the comment thread from the view' do
+            expect(response.body).to be_empty
+          end
+        end
+
+        context 'and having a reply' do
+          let(:commentable) { root_comment.commentable }
+          let(:root_reply) { create(:comment_request, commentable: commentable, body: 'This is a reply', parent_id: root_comment.id) }
```

Comment:
We should use `let!(...)` to avoid calling `root_reply` in the `before` block. It will still work as expected.

Reply:
You can even make this more clean by getting `root_comment.id` through `root_reply.commentable.id`, the you don't have to do anything in the before block but calling the route...

Reply:
Changed